grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationshipAtom.java

```diff
@@ -102,48 +103,62 @@
+
+    abstract ImmutableList<RelationPlayer> getRelationPlayers();
+    abstract ImmutableSet<Label> getRoleLabels();
+
+    private ImmutableList<Type> possibleTypes = null;
+
+    public static RelationshipAtom create(VarPattern pattern, Var predicateVar, @Nullable IdPredicate predicate, ReasonerQuery parent) {
         List<RelationPlayer> rps = new ArrayList<>();
-        getPattern().admin()
+        pattern.admin()
                 .getProperty(RelationshipProperty.class)
                 .ifPresent(prop -> rps.addAll(prop.relationPlayers()));
-        this.relationPlayers = ImmutableList.copyOf(rps);
-        this.roleLabels = ImmutableSet.<Label>builder().addAll(
+        ImmutableList<RelationPlayer> relationPlayers = ImmutableList.copyOf(rps);
+        ImmutableSet<Label> roleLabels = ImmutableSet.<Label>builder().addAll(
                 relationPlayers.stream()
                         .map(RelationPlayer::getRole)
                         .flatMap(CommonUtil::optionalToStream)
                         .map(VarPatternAdmin::getTypeLabel)
                         .flatMap(CommonUtil::optionalToStream)
                         .iterator()
         ).build();
+        RelationshipAtom relationshipAtom = new AutoValue_RelationshipAtom(pattern.admin().var(), pattern, predicateVar, predicate, relationPlayers, roleLabels);
+        relationshipAtom.parent = parent;
+        return relationshipAtom;
     }
 
-    private RelationshipAtom(VarPatternAdmin pattern, Var predicateVar, @Nullable IdPredicate predicate, ImmutableList<Type> possibleRelations, ReasonerQuery par) {
-        this(pattern, predicateVar, predicate, par);
-        this.possibleRelations = possibleRelations;
+    private static RelationshipAtom create(RelationshipAtom a, ReasonerQuery parent) {
+        RelationshipAtom relationshipAtom = new AutoValue_RelationshipAtom(a.getVarName(), a.getPattern(), a.getPredicateVariable(), a.getTypePredicate(), a.getRelationPlayers(), a.getRoleLabels());
+        relationshipAtom.parent = parent;
+        return relationshipAtom;
     }
 
-    private RelationshipAtom(RelationshipAtom a) {
-        super(a);
-        this.relationPlayers = a.relationPlayers;
-        this.roleLabels = a.roleLabels;
-        this.roleVarMap = a.roleVarMap;
-        this.possibleRelations = a.possibleRelations;
+    @Override
+    public Class<? extends VarProperty> getVarPropertyClass(){ return RelationshipProperty.class;}
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null || this.getClass() != obj.getClass()) return false;
+        if (obj == this) return true;
+        RelationshipAtom a2 = (RelationshipAtom) obj;
+        return Objects.equals(this.getTypeId(), a2.getTypeId())
+                && (isUserDefined() == a2.isUserDefined())
+                && getVarNames().equals(a2.getVarNames())
+                && getRelationPlayers().equals(a2.getRelationPlayers());
```

Comment:
We can add a comment here like you did above, so we know which properties are deliberately ignored.